### PR TITLE
CoffeeScript 2

### PR DESCRIPTION
**Pull request by jalavosus**
Basic changes to (hopefully) support coffeescript 2.

Essentially hoping that it doesn't require much besides requiring "coffeescript" instead of "coffee-script" and package.json requiring "coffeescript" at any version.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var coffee = require("coffee-script");
+var coffee = require("coffeescript");
```
Comment (michael-ciniawsky) [inline index.js]: wrap in a `try/catch` ? 😛

**Comment by d3viant0ne** [comment]
@jalavosus - This is going to take some internal discussion on how we are going to handle this. Give us a day or two and we will get you sorted out one way or another.

**Comment by jalavosus** [comment]
I'll take a look into the try-catch option when I get back to my computer. 

Can the package.json require both the old and new versions to be installed separately?

**Comment by michael-ciniawsky** [comment]
https://github.com/jashkenas/coffeescript/issues/4462

**Comment by michael-ciniawsky** [comment]
@jalavosus It's safe to just go with `coffeescript` v1 >= 1.8.x will be included in the new package and v2 is at `coffeescript@beta`

**Comment by jalavosus** [comment]
If I read that right, this (ostensibly) means blindly using `coffeescript`
is fine, yeah?

On Mar 13, 2017 13:02, "Michael Ciniawsky" <notifications@github.com> wrote:

jashkenas/coffeescript#4462
<https://github.com/jashkenas/coffeescript/issues/4462>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/webpack-contrib/coffee-loader/pull/30#issuecomment-286172934>,
or mute the thread

File: index.js (hunk 1 above)
Comment (bebraw) [inline index.js]: No need. Peer dep should force it. And it's better to fail hard in this case most likely.

**Comment by michael-ciniawsky** [comment]
@jalavosus I'm waiting for final confirmation in https://github.com/jashkenas/coffeescript/issues/4462, but yes setting `coffeescript` as `peerDependency` and `require('coffeescript')` should be fine, no rush you can also wait for confirmation, if you don't want to spend effort on it without 😛

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
 		"loader-utils": "^1.0.2"
 	},
 	"peerDependencies": {
-		"coffee-script": "1.x"
+	    "coffeescript": "*"
```
Comment (bebraw) [inline package.json]: The risk with this is that if they break their API in a major, the loader will break. So change to a range if you want to be careful.

Comment (michael-ciniawsky) [inline package.json]: Yep, we need to test for breaking Node API changes before release

```js
{
  "peerDependencies": {
     "coffeescript": ">= 1.8.x"
   }
}
```

**Comment by michael-ciniawsky** [comment]
Closes #33

**Comment by jalavosus** [comment]
Updated package.json, sorry for the holdup.

**Pull request merged**